dev/generate_bom.py

```diff
@@ -125,6 +127,38 @@ def write_docker_version_files(self):
       with open(config_file, 'w') as cfg:
         cfg.write(docker_tag)
 
+  def generate_changelog(self):
+    """Generate a release changelog and write it to a file.
+
+    The changelog contains a section per microservice that describes the
+    changes made since the last Spinnaker release. It also contains the
+    version information as well.
+    """
+    changelog = 'Spinnaker {0}\n\n'.format(self.__toplevel_version)
+    for comp, hash in self.__changelog_start_hashes.iteritems():
+      version_bump = self.__component_versions[comp]
+      next_tag_with_build = '{0}-{1}'.format(version_bump.version_str,
+                                             self.build_number)
+      first_dash_idx = next_tag_with_build.index('-')
+      gradle_version = next_tag_with_build[first_dash_idx + 1:]
+
+      # Generate the changelog for the component.
+      print 'Generating changelog for {comp}...'.format(comp=comp)
+      result = run_quick('cd {comp}; clog -f {hash} --setversion {version}; cd ..'
+                         .format(comp=comp, hash=hash, version=gradle_version))
+      if result.returncode != 0:
+        print "Changelog generation failed for {0} with \n{1}\n exiting...".format(comp, result.stdout)
+        exit(result.returncode)
+      # Capitalize
+      comp_cap = comp[0].upper() + comp[1:]
+      changelog += '\n# {0}\n{1}'.format(comp_cap, result.stdout)
+    print 'Writing changelog...'
+    # Write the changelog with the toplevel version without the build number.
+    # This is ok since the changelog is only published if the toplevel version is released.
+    changelog_file = '{0}-changelog.md'.format(self.__toplevel_version)
+    with open(changelog_file, 'w') as clog:
```

Comment:
Do you mean specify the file or the output path with the file name as I have it now?